src/transformStateWithClones.js

```diff
@@ -7,7 +7,28 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const stateClonesArray = [];
+  const stateCopyObject = { ...state };
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        Object.assign(stateCopyObject, action.extraData);
+        break;
+      case 'removeProperties':
+        action.keysToRemove.forEach((key) => delete stateCopyObject[key]);
+        break;
+      case 'clear':
+        Object.keys(stateCopyObject).forEach(
+          (key) => delete stateCopyObject[key]
+        );
+        break;
+    }
```

Comment:
Return default case but instead of string [throw error](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/throw?retiredLocale=uk) according previous review comment